rustworkx-core/src/centrality.rs

```diff
@@ -1143,3 +1132,420 @@ where
+        }
+    }
+    closeness
+}
+
+#[cfg(test)]
+mod test_newman_weighted_closeness_centrality {
+    use crate::centrality::closeness_centrality;
+
+    use super::newman_weighted_closeness_centrality;
+    use petgraph::visit::EdgeRef;
+
+    macro_rules! assert_almost_equal {
+        ($x:expr, $y:expr, $d:expr) => {
+            if ($x - $y).abs() >= $d {
+                panic!("{} != {} within delta of {}", $x, $y, $d);
+            }
+        };
+    }
+
+    #[test]
+    fn test_weighted_closeness_graph() {
+        let g = petgraph::graph::UnGraph::<u32, f64>::from_edges([
+            (0, 1, 1.0),
+            (1, 2, 1.0),
+            (2, 3, 1.0),
+            (3, 4, 1.0),
+            (4, 5, 1.0),
+            (5, 6, 1.0),
+        ]);
+        let classic_closeness = closeness_centrality(&g, false);
+        let weighted_closeness = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+
+        assert_eq!(classic_closeness, weighted_closeness);
+    }
+
+    #[test]
+    fn test_the_same_as_closeness_centrality_when_weights_are_1_not_improved_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (0, 1, 1.0),
+            (1, 2, 1.0),
+            (2, 3, 1.0),
+            (3, 4, 1.0),
+            (4, 5, 1.0),
+            (5, 6, 1.0),
+        ]);
+        let classic_closeness = closeness_centrality(&g, false);
+        let weighted_closeness = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+
+        assert_eq!(classic_closeness, weighted_closeness);
+    }
+
+    #[test]
+    fn test_the_same_as_closeness_centrality_when_weights_are_1_improved_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (0, 1, 1.0),
+            (1, 2, 1.0),
+            (2, 3, 1.0),
+            (3, 4, 1.0),
+            (4, 5, 1.0),
+            (5, 6, 1.0),
+        ]);
+        let classic_closeness = closeness_centrality(&g, true);
+        let weighted_closeness = newman_weighted_closeness_centrality(&g, true, |x| *x.weight());
+
+        assert_eq!(classic_closeness, weighted_closeness);
+    }
+
+    #[test]
+    fn test_weighted_closeness_two_connected_components_not_improved_digraph() {
+        let g = petgraph::graph::DiGraph::<u32, f64>::from_edges([
+            (0, 1, 1.0),
+            (1, 2, 0.5),
+            (2, 3, 0.25),
+            (4, 5, 1.0),
+            (5, 6, 0.5),
+            (6, 7, 0.25),
+        ]);
+        let c = newman_weighted_closeness_centrality(&g, false, |x| *x.weight());
+        let result = [
+            Some(0.0),
+            Some(1.0),
+            Some(0.4),
+            Some(0.176470),
+            Some(0.0),
+            Some(1.0),
+            Some(0.4),
+            Some(0.176470),
+        ];
+
+        for i in 0..8 {
+            assert_almost_equal!(result[i].unwrap(), c[i].unwrap(), 1e-4);
+        }
```

Comment:
You don't need to hardcode the length of the vector if you iterate with `zip`
```suggestion
        for (&expected, &computed) in result.iter().zip(c.iter()) {
             assert_almost_equal!(expected.unwrap(), computed.unwrap(), 1e-4);
         }
```

Bonus points if you make it a macro